com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -607,6 +607,14 @@ public void SetRelayServerData(string ipv4Address, ushort port, byte[] allocatio
             SetProtocol(ProtocolType.RelayUnityTransport);
         }
 
+        /// <summary>Set the relay server data (using the lower-level Unity Transport data structure).</summary>
+        /// <param name="serverData">Data for the Relay server to use.</param>
+        public void SetRelayServerData(RelayServerData serverData)
+        {
+            m_RelayServerData = serverData;
+            SetProtocol(ProtocolType.RelayUnityTransport);
```

Comment:
Personally, I would rather have two APIs. One to enable/set Relay protocol and one to feed it the `ServerData`. If we elect to set both with the same function call, then, I'd still prefer an explicit boolean to state if I want to enable the Relay protocol.

I realize it's useless to set the RelayServerData if you're not using the Relay protocol. Still, we don't know in what order the game code will receive the bits and pieces it needs, or it wants to switch protocols back and forth. So, I'd follow the single responsibility principle and have protocol selection separate from setting the relay server data.

Reply:
I agree with you in principle, but unfortunately I think that ship has already sailed. The existing `SetRelayServerData`, `SetClientRelayData`, and `SetHostRelayData` methods all automatically set the protocol at the same time as the server data. Similarly, `SetConnectionData` also sets the protocol automatically.

If we change this now, we're likely to break existing code that relies on the current behavior. And while the new API could opt not to set the protocol, it would create an inconsistency with the other APIs. I think the `UnityTransport` API is already confusing as it is.

(Also if we were to address this, I'd personally do away with the protocol type entirely. Using Relay or not should just be a binary toggle like we have for WebSockets and encryption.)

Reply:
yeah, makes total sense, then. Consistency first.